### DISCUSSION TITLE
armbian linux arm64 build failed

### DISCUSSION THREAD

**Discussion by toplinuxsir** [rust-skia/rust-skia#831] [Q&A] [answered]
linux arm64 build failed
```
SkMath.h:14:10: fatal error: 'cstdint' file not found
  #include <cstdint>

SkTypes.h:26:10: fatal error: 'climits' file not found
  #include <climits>
```

**Comment by pragmatrix** [comment]
I've just built the arm64 Dockerfile in this repository on macOS M2 successfully: https://github.com/rust-skia/rust-skia/blob/master/docker/linux/arm64/Dockerfile. I suspect that g++ or clang is missing from your setup. Please verify that it has the packages (or equivalent ones) installed that are listed in the Dockerfile.

If you attempt to cross-compile, things get a bit harder to set up, you might want to take a look at #757 and perhaps install additional packages like `gcc-aarch64-linux-gnu`, and `libstdc++-10-dev-arm64-cross`.
 
If that does not work, please provide a Dockerfile or more information about the exact Linux OS version you want to build rust-skia on.

**Comment by toplinuxsir** [comment]
> https://github.com/rust-skia/rust-skia/blob/master/docker/linux/arm64/Dockerfile

I have already installed all the dependecy libs, Still not found the cstdint and climits

The os : armbian 23.5 jammy

**Comment by pragmatrix** [comment]
As long there is no Dockerfile to reproduce the problem, I am unable to help here. What you can do is to take a look where the C++ includes are and see if the include directories are set. In #757 you can find an example on how to modify the include directories (in `skia-bindings/build_support/platform`).

**Comment by toplinuxsir** [comment]
@pragmatrix  when build with skia-binding how to specify the include directory ?
Thanks!

**Comment by toplinuxsir** [comment]
@pragmatrix cstdint climits is c++ standard include files , How to specify the include directory when build with cargo command ? Thanks

**Comment by pragmatrix** [comment]
@toplinuxsir The environment variable `CPLUS_INCLUDE_PATH` should work for clang and g++.

**Comment by toplinuxsir** [comment, marked as answer]
@pragmatrix  Thanks , I use the below  command to build successfully under armbian linux , Hope this help others.
```
CPLUS_INCLUDE_PATH="/usr/include/c++/11:/usr/include/aarch64-linux-gnu/c++/11" cargo build --release
```
